src/hooks/fetch.ts

```diff
@@ -0,0 +1,397 @@
+   * All the additional parameters are extracted from the extender context object.
+   * @param {RequestExtenderContext} args
+   * @param {string} args.transactionId The transaction id of the request
+   * @param {string} args.awsRequestId The AWS request ID of the current lambda invocation
+   * @param {RequestData} args.requestData The request data object
+   * @param {string} args.requestRandomId The random ID of the request
+   * @param {Response} args.response The response object returned by the fetch promise
+   * @private
+   */
+  private static async createResponseSpan({
+    transactionId,
+    awsRequestId,
+    requestData,
+    requestRandomId,
+    response,
+  }: RequestExtenderContext): Promise<void> {
+    if (!response) {
+      return;
+    }
+
+    const clonedResponse = response.clone();
+    const responseData = FetchInstrumentation.convertResponseToResponseData(clonedResponse);
+    const responseDataWriterHandler = BaseHttp.createResponseDataWriterHandler({
+      transactionId,
+      awsRequestId,
+      requestData,
+      requestRandomId,
+      response: responseData,
+    });
+
+    const bodyText = await clonedResponse.text();
+    responseDataWriterHandler(['data', bodyText]);
+    responseDataWriterHandler(['end']);
+  }
+
+  /**
+   * Parses the raw arguments passed to the fetch function and returns the URL and options object.
+   * @param {FetchArguments} args The raw fetch arguments
+   * @param {RequestInfo | URL} args.input The first argument (called input / resource) that is passed to the fetch command
+   * @param {RequestInit} args.init The second argument (called init / options) that is passed to the fetch command
+   * @returns {UrlAndRequestOptions} Our custom request options object containing the URL and options
+   * @private
+   */
+  private static async parseRequestArguments({
+    input,
+    init,
+  }: FetchArguments): Promise<FetchUrlAndRequestOptions> {
+    let url: string = undefined;
+    const options: ParseHttpRequestOptions & {
+      body?: string;
+    } = {
+      headers: {},
+      method: 'GET',
+    };
+
+    if (input instanceof URL) {
+      url = input.toString();
+    } else if (typeof input === 'string') {
+      url = input;
+    } else if (input instanceof Request) {
+      url = input.url;
+      options.method = input.method || 'GET';
+      options.headers = FetchInstrumentation.convertHeadersToKeyValuePairs(input.headers);
+    }
+
+    if (init) {
+      options.method = init.method || options.method || 'GET';
+      options.headers = {
+        ...options.headers,
+        ...FetchInstrumentation.convertHeadersToKeyValuePairs(init.headers),
+      };
+    }
+
+    // Read the body from the request object, only if we shouldn't look in the init object
+    let body: string = undefined;
+    try {
+      if (input && input instanceof Request && input.body && !init && !init?.body) {
```

Comment:
You can abbreviate it to:
```suggestion
      if (input instanceof Request && input?.body && !init?.body) {
```